test_app/tests.py

```diff
@@ -129,8 +129,12 @@ def test_limit_to_choice_for_chained_manytomany(self):
     # grouped foreignkey
     def test_student_add_get(self):
         response = self.client.get(reverse('admin:test_app_student_add'))
-        self.assertContains(response, '<optgroup label="Grade 1">\n<option value="1">   Team 1</option>\n</optgroup>')
-        self.assertContains(response, '<optgroup label="Grade 2">\n<option value="2">   Team 2</option>\n</optgroup>')
+        # For some reason, Django 1.11 renders this with additional new lines and spaces, so we strip all of them off
+        splitted = response.content.split(b'\n')
+        splitted = [part.lstrip(b' \t\n\r') for part in splitted]
+        response.content = b''.join(splitted)
+        self.assertContains(response, '<optgroup label="Grade 1"><option value="1">   Team 1</option></optgroup>')
```

Comment:
Instead of doing all of that work yourself, just add `html=True` to the calls to `assertContains` and let the assertion html HTML differences for you.